src/app/tabs/run-tab.js

```diff
@@ -581,6 +584,52 @@ function settings (container, self) {
       }
     })
   }
+  function signMessage (event) {
+    self._deps.udapp.getAccounts((err, accounts) => {
+      if (err) { addTooltip(`Cannot get account list: ${err}`) }
+      var signMessageDialog = {'title': 'Sign a message', 'text': 'Enter a message to sign', 'inputvalue': 'Message to sign' }
+      var $txOrigin = container.querySelector('#txorigin')
+      var account = $txOrigin.selectedOptions[0].value
+      var isVM = executionContext.isVM()
+      if (isVM) {
+        modalDialogCustom.promptMulti(signMessageDialog, (message) => {
+          const personalMsg = ethJSUtil.hashPersonalMessage(Buffer.from(message))
+          var privKey = self._deps.udapp.accounts[account].privateKey
+          try {
+            var rsv = ethJSUtil.ecsign(personalMsg, privKey)
+            var rsvJson = JSON.stringify(rsv, null, '\t')
+            var signedData = ethJSUtil.toRpcSig(rsv.v, rsv.r, rsv.s)
+            modalDialogCustom.alert(rsvJson + '\n' + signedData)
+          } catch (e) {
+            addTooltip(e.message)
+            return
+          }
+        }, false)
+      } else {
+        modalDialogCustom.promptPassphrase('Passphrase to sign a mesasge', 'Enter your passphrase for this account to sign the message', 'Passphrase', (passphrase) => {
+          modalDialogCustom.promptMulti(signMessageDialog, (message) => {
+            console.log(executionContext.web3())
+            const hashedMsg = executionContext.web3().sha3(message)
+            try {
+              var personal = new Personal(executionContext.web3().currentProvider)
+              personal.sign('0x' + Buffer.from(message).toString('hex'), account, passphrase, (error, signedData) => {
+                if(error && erorr.message !== '') {
+                  console.log(error)    
+                  addTooltip(error.message)
+                }
+                modalDialogCustom.alert(signedData)
```

Comment:
@yann300 Could you review here

Reply:
no need to alert here if there's an error

Reply:
Fixed!